agreement/msgp_gen.go

```diff
@@ -0,0 +1,3958 @@
+					if err != nil {
+						err = msgp.WrapError(err, "Proposals", zb0002)
+						return
+					}
+				}
+			default:
+				err = msgp.ErrNoField(string(field))
+				if err != nil {
+					err = msgp.WrapError(err)
+					return
+				}
+			}
+		}
+	}
+	o = bts
+	return
+}
+
+func (_ *equivocationVoteAuthenticator) CanUnmarshalMsg(z interface{}) bool {
+	_, ok := (z).(*equivocationVoteAuthenticator)
+	return ok
+}
+
+// Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
+func (z *equivocationVoteAuthenticator) Msgsize() (s int) {
+	s = 1 + 4 + (*z).Sender.Msgsize() + 5 + (*z).Cred.Msgsize() + 4 + msgp.ArrayHeaderSize
+	for zb0001 := range (*z).Sigs {
+		s += (*z).Sigs[zb0001].Msgsize()
+	}
+	s += 6 + msgp.ArrayHeaderSize
+	for zb0002 := range (*z).Proposals {
+		s += (*z).Proposals[zb0002].Msgsize()
+	}
+	return
+}
+
+// MsgIsZero returns whether this is a zero value
+func (z *equivocationVoteAuthenticator) MsgIsZero() bool {
+	return ((*z).Sender.MsgIsZero()) && ((*z).Cred.MsgIsZero()) && (((*z).Sigs[0].MsgIsZero()) && ((*z).Sigs[1].MsgIsZero())) && (((*z).Proposals[0].MsgIsZero()) && ((*z).Proposals[1].MsgIsZero()))
+}
+
+// MarshalMsg implements msgp.Marshaler
+func (z period) MarshalMsg(b []byte) (o []byte, err error) {
+	o = msgp.Require(b, z.Msgsize())
+	o = msgp.AppendUint64(o, uint64(z))
+	return
+}
+
+func (_ period) CanMarshalMsg(z interface{}) bool {
+	_, ok := (z).(period)
+	return ok
+}
+
+// UnmarshalMsg implements msgp.Unmarshaler
+func (z *period) UnmarshalMsg(bts []byte) (o []byte, err error) {
+	{
+		var zb0001 uint64
+		zb0001, bts, err = msgp.ReadUint64Bytes(bts)
+		if err != nil {
+			err = msgp.WrapError(err)
+			return
+		}
+		(*z) = period(zb0001)
+	}
+	o = bts
+	return
+}
+
+func (_ *period) CanUnmarshalMsg(z interface{}) bool {
+	_, ok := (z).(*period)
+	return ok
+}
+
+// Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
+func (z period) Msgsize() (s int) {
+	s = msgp.Uint64Size
+	return
+}
+
+// MsgIsZero returns whether this is a zero value
+func (z period) MsgIsZero() bool {
+	return z == 0
+}
+
+// MarshalMsg implements msgp.Marshaler
+func (z *proposal) MarshalMsg(b []byte) (o []byte, err error) {
+	o = msgp.Require(b, z.Msgsize())
+	// omitempty: check for empty values
+	zb0001Len := uint32(4)
+	var zb0001Mask uint8 /* 6 bits */
+	if (*z).unauthenticatedProposal.Block.MsgIsZero() {
+		zb0001Len--
+		zb0001Mask |= 0x1
+	}
+	if (*z).unauthenticatedProposal.OriginalPeriod == 0 {
+		zb0001Len--
+		zb0001Mask |= 0x4
+	}
+	if (*z).unauthenticatedProposal.OriginalProposer.MsgIsZero() {
+		zb0001Len--
+		zb0001Mask |= 0x8
+	}
+	if (*z).unauthenticatedProposal.SeedProof.MsgIsZero() {
+		zb0001Len--
+		zb0001Mask |= 0x10
+	}
+	// variable map header, size zb0001Len
+	o = append(o, 0x80|uint8(zb0001Len))
+	if zb0001Len != 0 {
+		if (zb0001Mask & 0x1) == 0 { // if not empty
+			// string "Block"
```

Comment:
It looks like `bookkeeping.Block` is not embedded properly in the generated code.

Reply:
Ah, I see, it doesn't know that it's an embedded struct, because it's from another package..

Will think about how to fix this.

Reply:
If you update msgp to version `v1.1.37`, it should correctly handling embedding of struct types from imported packages, like `bookkeeping.Block`.

Reply:
I can confirm that `v1.1.37` correctly handles the embedding of `bookkeeping.Block`. Thanks!